openfisca_core/entities.py

```diff
@@ -51,9 +51,21 @@ def check_role_validity(self, role):
         if role is not None and not type(role) == Role:
             raise Exception("{} is not a valid role".format(role))
 
-    def __call__(self, variable_name, period, options = [], **parameters):
+    def __call__(self, variable_name, period=None, options = [], **parameters):
         self.check_variable_defined_for_entity(variable_name)
 
+        if period is None:
+            import traceback
```

Comment:
For a better separation of concerns and to keep `__call__` readable, I'd handle the error in a separate function (`check_something`, or `handle_something`).